core/modules/layout/layout.module

```diff
@@ -979,15 +978,53 @@ function layout_load_multiple($layout_names, $skip_menu_items = FALSE) {
   return $layouts;
 }
 
+/**
+ * Load all layouts for a given router item and allow other modules to alter
+ * them and/or set their contexts.
+ *
+ * @param array $router_item
+ *   The fully built menu router item, e.g., what is returned by menu_get_item().
+ * @param bool $skip_menu_items
+ *   Flag to skip the attempted loading of menu items for the loaded layouts.
+ *
+ * @return Layout[]
+ *   An array of load Layout object instances.
+ */
+function layout_load_multiple_by_router_item($router_item, $skip_menu_items = NULL) {
+  $href = $router_item['href'];
+  $layouts_by_item = &backdrop_static(__FUNCTION__, array());
+  if (isset($layouts_by_item[$href])) {
+    return $layouts_by_item[$href];
+  }
+  $layouts = layout_load_multiple_by_path($router_item['path'], $skip_menu_items);
+  backdrop_alter('layout_layouts', $layouts, $router_item);
```

Comment:
After reading through this PR, I think this hook should be renamed. This hook does not apply when using `layout_load()`, `layout_load_multiple()`, or `layout_load_by_path()`, so I worry that it is going to be confusing based on its name implying that it would be called in those situations. The documentation you added helps, but the name itself is misleading.

Instead, I would suggest:
```suggestion
  backdrop_alter('layout_load_by_router_item_alter', $layouts, $router_item);
```

Which would make the hook name `hook_layout_load_by_router_item_alter()`. 

While it is a long hook name, it's not unusual. We have other long hook names like `hook_field_formatter_settings_summary_alter()`, `hook_block_view_MODULE_NAME_DELTA_alter()`, and `hook_form_FORM_ID_alter()`.

This also makes this name similar to `hook_redirect_load_by_source_alter()`, which has the same sort of situation where it's only called in certain loading situations.

Reply:
Hook named change to your suggestion in update to the PR.